internal/config/config.go

```diff
@@ -370,9 +372,23 @@ func peerFromCR(p metallbv1beta2.BGPPeer) (*Peer, error) {
 		nodeSels = []labels.Selector{labels.Everything()}
 	}
 
+	if p.Spec.Password != "" && p.Spec.PasswordSecret.Name != "" {
+		return nil, fmt.Errorf("can not have both password and secret ref set in peer config %q/%q", p.Namespace,
+			p.Name)
+	}
+
 	var password string
 	if p.Spec.Password != "" {
 		password = p.Spec.Password
+	} else if p.Spec.PasswordSecret.Name != "" {
+		var secret corev1.Secret
+		var ok bool
+		if secret, ok = passwordSecrets[p.Spec.PasswordSecret.Name]; !ok {
```

Comment:
I'd rather have the := form and checking if !ok on the next line
secret, ok: = ...
if !ok {
}